src/solvers/smt2/smt2_dec.cpp

```diff
@@ -140,7 +140,7 @@ decision_proceduret::resultt smt2_dect::dec_solve()
             + smt2_temp_file.temp_result_filename;
     break;
 
-  default:
+  case solvert::GENERIC:
     assert(false);
```

Comment:
If we are changing the default behaviour, can we do something better than just assert(false)?